pkg/router/template/router.go

```diff
@@ -877,28 +880,49 @@ func getServiceUnits(route *routeapi.Route) map[string]int32 {
 
 	// get the weight and number of endpoints for each service
 	key := endpointsKeyFromParts(route.Namespace, route.Spec.To.Name)
-	serviceUnits[key] = 0
-	if route.Spec.To.Weight != nil {
-		serviceUnits[key] = int32(*route.Spec.To.Weight)
-	}
-	if serviceUnits[key] < 0 || serviceUnits[key] > 256 {
-		serviceUnits[key] = 0
-	}
+	serviceUnits[key] = getServiceUnitWeight(route.Spec.To.Weight)
 
 	for _, svc := range route.Spec.AlternateBackends {
 		key = endpointsKeyFromParts(route.Namespace, svc.Name)
-		serviceUnits[key] = 0
-		if svc.Weight != nil {
-			serviceUnits[key] = int32(*svc.Weight)
-		}
-		if serviceUnits[key] < 0 || serviceUnits[key] > 256 {
-			serviceUnits[key] = 0
-		}
+		serviceUnits[key] = getServiceUnitWeight(svc.Weight)
 	}
 
 	return serviceUnits
 }
 
+// getServiceUnitWeight takes a reference to a weight and returns its value or the default.
+// It also checks that it is in the correct range.
+func getServiceUnitWeight(weightRef *int32) int32 {
+	// Default to 1 if there is no weight
+	var weight int32 = 1
+	if weightRef != nil {
+		weight = *weightRef
+	}
+
+	// Do a bounds check
+	if weight < 0 {
+		weight = 0
+	} else if weight > 256 {
+		weight = 256
```

Comment:
When weight >= 256, don't we want to set it to 0?
Otherwise in Line 918, we count this serviceUnit toward active endpoints which doesn't seem right.

Reply:
Assuming we want to ignore the service unit if the corresponding weight is invalid.

Reply:
weight=0 clearly means user want to ignore the endpoint but weight > 256 means mostly some typo/calculation error and may not mean ignore the endpoint.

It may be useful to log a warning when we are defaulting invalid weights to 0 or 256?